src/main/java/org/opensearch/knn/index/codec/KNN990Codec/QuantizationConfigKNNCollector.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.KNN990Codec;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.lucene.search.KnnCollector;
+import org.apache.lucene.search.TopDocs;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+
+/**
+ * Collector used for passing the quantization state during query flow.
+ */
+@Setter
+@Getter
+public class QuantizationConfigKNNCollector implements KnnCollector {
+
+    private QuantizationState quantizationState;
+
+    @Override
+    public boolean earlyTerminated() {
+        throw new UnsupportedOperationException("Search functionality using codec is not supported with Native Engine Reader");
```

Comment:
Can we create a constant for this error message in this class and use it in all the methods ?